### PR TITLE
add circleci 2.0

### PR DESCRIPTION
**Pull request by szilvesztersimicza**
The CircleCI 1.0 is no longer available. So this is a migration PR to upgrade to CircleCI 2.0.
https://circleci.com/sunset1-0/

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,114 @@
+version: 2.0
+
+install: &install
```
Comment (richm) [inline .circleci/config.yml]: what does the `&install` here mean? I'm not familiar with this syntax

File: .circleci/config.yml [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+      - run: *missingdeps
+      - checkout
+      - run: *install
+      - run: *test
+      - store_test_results:
+            path: coverage
+      - store_artifacts:
+            path: coverage
+
+  "ruby-2.3.8":
```
Comment (richm) [inline .circleci/config.yml]: Do we really need to test ruby 2.3 and earlier?  What major platforms still require ruby 2.3 and earlier?

File: .circleci/config.yml (hunk 1 above)
Comment (szilvesztersimicza) [reply .circleci/config.yml]: It is like a reference or label, so defined once and used many times.

File: .circleci/config.yml (hunk 2 above)
Comment (szilvesztersimicza) [reply .circleci/config.yml]: It was in the  circleci 1.0 config file, just migrated them to the 2.0 config.
We will be more than happy throw old ruby versions away.

**Comment by szilvesztersimicza** [comment]
I have done some updates. Please Review it and if its correct please merge it! 
Thank You!

**Pull request merged**